File: core/Classes/Data/Service/DTO/AuthAndSessions/Session/CreateFirstSessionJson.swift
//
//  CreateFirstSessionJson.swift
//  core
//
//  Created by Manu Herrera on 24/04/2020.
//

import Foundation

struct CreateFirstSessionJson: Codable {

    let client: ClientJson
    let gcmToken: String
    let primaryCurrency: String
    let basePublicKey: PublicKeyJson

}

struct CreateFirstSessionOkJson: Codable {

    let user: User
    let cosigningPublicKey: PublicKeyJson
    let swapServerPublicKey: PublicKeyJson

}

struct ClientJson: Codable {

    let type: String
    let buildType: String
    let version: Int

}

File: libwallet/service/model/create_first_session_json.go
package model

type CreateFirstSessionJSON struct {
	Client          ClientJSON    `json:"client"`
	GcmToken        *string       `json:"gcmToken,omitempty"`
	PrimaryCurrency string        `json:"primaryCurrency"`
	BasePublicKey   PublicKeyJSON `json:"basePublicKey"`
}
